lib/modules/manager/pip-compile/artifacts.ts

```diff
@@ -108,12 +114,22 @@ export async function updateArtifacts({
       const compileArgs = extractHeaderCommand(existingOutput, outputFileName);
       const cwd = inferCommandExecDir(outputFileName, compileArgs.outputFile);
       const upgradePackages = updatedDeps.filter((dep) => dep.isLockfileUpdate);
-      const packageFile = pipRequirements.extractPackageFile(newInputContent);
+      const packageFiles: PackageFileContent[] = [];
+      for (const name of compileArgs.sourceFiles) {
+        const path = upath.join(cwd, name);
+        const content = await readLocalFile(path, 'utf8');
+        if (content) {
+          const packageFile = extractPackageFile(content, path, {});
```

Comment:
seems to be expensive, can't we split the extract function to smaller part's, so we only extract the relevant source urls instead of all dependencies again?